test/e2e/serial/workload_placement_test.go

```diff
@@ -151,3 +165,94 @@ func filterNodesWithMoreThanOneNUMAZone(nodes []corev1.Node, nodeResourceTopolog
 
 	return ret, nil
 }
+
+func getWorkloadRequirements(nodeResourceTopologies []nrtv1alpha1.NodeResourceTopology, nodes []corev1.Node) (corev1.ResourceList, error) {
+	for _, nrt := range nodeResourceTopologies {
+		// for all the zones
+		for _, zone := range nrt.Zones {
+			// find zones bigger than the current one
+			zonesMap, err := zonesBiggerThan(zone, nodeResourceTopologies)
+			if err != nil {
+				return nil, err
+			}
+			if len(zonesMap) == 0 {
+				// if no other zone matches but the zone itself.
+				// find if any node but the one this zone comes from can match the requests
+				for _, node := range nodes {
+					if node.Name == nrt.Name {
+						continue
+					}
+					//convert zone resources into a resource request
+					zoneResources := resourceInfoListToResourceList(&zone.Resources)
```

Comment:
why are we passing the zone resources by pointer?

Reply:
premature ( and unnecessary ) optimization is the root of all evil ... just to avoid copy

Reply:
let's avoid that indeed. In this specific case, you're copying a `map` anyway.